### DISCUSSION TITLE
Development Environment Recommendations

### DISCUSSION THREAD

**Discussion by jmelkun** [magento/pwa-studio#3163] [Q&A]
Hello all,

I've been using [Warden](https://warden.dev) to create local development environments for Magento 2, but I've been struggling to get it to work with PWA Studio on my WIndows laptop. Does anyone have a local dev environment they'd recommend for use with PWA Studio?

Thanks!

**Comment by sirugh** [comment]
Have you tried WSL on Windows?

**Reply by jmelkun** [reply]
I have used WSL, but when using it with my Docker environment the URL doesn't resolve properly -- instead of broadcasting to 0.0.0.0 it goes to 127.0.0.1 and can only be accessed as http://localhost:3000 instead of the generated URL. I've tried using netsh to bridge the connection and it kind of works, but it seems pretty far from ideal
